src/libraries/System.Private.CoreLib/src/System/Text/ASCIIUtility.cs

```diff
@@ -240,7 +241,14 @@ private static unsafe nuint GetIndexOfFirstNonAsciiByte_Sse2(byte* pBuffer, nuin
 
             // Read the first vector unaligned.
 
-            currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            if (Sse2.IsSupported)
+            {
+                currentMask = (uint)Sse2.MoveMask(Sse2.LoadVector128(pBuffer)); // unaligned load
+            }
+            else
+            {
+                currentMask = AdvSimd.Arm64.MaxAcross(AdvSimd.LoadVector128(pBuffer)).ToScalar();
```

Comment:
I could use the new `AdvSimd.MoveMask` method here. I'll refactor all these calls once a PR with that method goes in.